engine/src/main/scala/cromwell/engine/workflow/lifecycle/materialization/MaterializeWorkflowDescriptorActor.scala

```diff
@@ -180,18 +184,36 @@ object MaterializeWorkflowDescriptorActor {
         s"'$optionName' is specified in workflow options but value is not of expected Double type: ${e.getMessage}".invalidNel
     }
   }
+
+  def getImportAuthProviders(conf: Config,
+                             authProviderFunc: String => ImportAuthProvider
+  ): ErrorOr[List[ImportAuthProvider]] = {
+    val isPrivateWorkflowsEnabled = conf.as[Boolean]("private-workflows.enabled")
```

Comment:
I think there's a philosophical question here about whether 

1. The materializer should have a config that decides whether or not it opts-in to the auth provider that's configured
2. The materializer always opts in to the auth provider, which can be configured whether or not to supply auth tokens (and if so, what type)

I think my personal preference would be for 2, but it might be worth considering the pros and cons of each

Reply:
FWIW:
-  I think the auth vending service is already configurable to opt-out of doing any auth lookups
- It would be nice if the describe and execution were configured in the same place and in the same way
- It might make the logic here a little simpler?

Reply:
I like the idea of the describe and execution functions referencing the same config, and of all the config checking happening inside `GithubAuthVendingSupport`.

Reply:
>The materializer always opts in to the auth provider, which can be configured whether or not to supply auth tokens (and if so, what type)

So in the `MaterializeWorkflowDescriptorActor` it would always pass in the GithAuthProvider to the HttpResolver and it GithAuthVending service will be opting in or out of doing auth lookups based on the config. Is my understanding right?